File: Basic/DotNet/AspNet/WaveData.cs
using System;
using System.Text;

namespace RestSample
{
    public class WaveData
    {
        #region Public Properties

        public double Cos { get; set; }
        public double Cosh { get; set; }
        public int Order { get; set; }

        public double Radians { get; set; }
        public double Sin { get; set; }
        public double Sinh { get; set; }
        public double Tan { get; set; }
        public double Tanh { get; set; }
        public double Tau { get; set; }

        #endregion Public Properties

        #region Public Methods

        public static WaveData Next(TimeSpan interval, double multiplier, int order)
        {
            var now = DateTime.UtcNow;
            var timeOfDay = now.TimeOfDay;

            var radians = ((timeOfDay.TotalMilliseconds % interval.TotalMilliseconds) / interval.TotalMilliseconds) * 2 *
                          Math.PI;

            return new WaveData
            {
                Order = order,
                Radians = radians,
                Tau = radians / (2 * Math.PI),
                Sin = multiplier * Math.Sin(radians),
                Cos = multiplier * Math.Cos(radians),
                Tan = multiplier * Math.Tan(radians),
                Sinh = multiplier * Math.Sinh(radians),
                Cosh = multiplier * Math.Cosh(radians),
                Tanh = multiplier * Math.Tanh(radians)
            };
        }

        public override string ToString()
        {
            var builder = new StringBuilder();
            builder.AppendLine($"Order = {Order}");
            builder.AppendLine($"Radians = {Radians}");
            builder.AppendLine($"Tau     = {Tau}");
            builder.AppendLine($"Sine    = {Sin}");
            builder.AppendLine($"Cosine  = {Cos}");
            builder.AppendLine($"Tangent = {Tan}");
            builder.AppendLine($"Sinh    = {Sinh}");
            builder.AppendLine($"Cosh    = {Cosh}");
            builder.AppendLine($"Tanh    = {Tanh}");
            return builder.ToString();
        }

        #endregion Public Methods
    }
}

File: ocs_samples/basic_samples/SDS/JavaScript/NodeJs/WaveData.js
// WaveData.js
//

module.exports = {
    // Wave Data Object
    WaveData: function() {
        this.Order = null;
        this.Radians = null;
        this.Tau = null;
        this.Sin = null;
        this.Cos = null;
        this.Tan = null;
        this.Sinh = null;
        this.Cosh = null;
        this.Tanh = null;
    },
    
    WaveDataCompound: function() {
        this.Order = null;
        this.Multiplier = null;
        this.Radians = null;
        this.Tau = null;
        this.Sin = null;
        this.Cos = null;
        this.Tan = null;
        this.Sinh = null;
        this.Cosh = null;
        this.Tanh = null;
    },

    // creates a new wave event
    NextWave: function(interval, multiplier, order) {
        
        radians = order * Math.PI/32 +1;

        newWave = new this.WaveData();
        newWave.Order = order;
        newWave.Radians = radians;
        newWave.Tau = radians / (2 * Math.PI);
        newWave.Sin = multiplier * Math.sin(radians);
        newWave.Cos = multiplier * Math.cos(radians);
        newWave.Tan = multiplier * Math.tan(radians);
        newWave.Sinh = multiplier * Math.sinh(radians);
        newWave.Cosh = multiplier * Math.cosh(radians);
        newWave.Tanh = multiplier * Math.tanh(radians);

        return newWave;
    },
    
    // creates a new wave event
    NextWaveCompound: function(order, multiplier) {
        radians = order * 2 * Math.PI;

        newWave = new this.WaveDataCompound();        
        newWave.Order = order;
        newWave.Multiplier = multiplier;
        newWave.Radians = radians;
        newWave.Tau = radians / (2 * Math.PI);
        newWave.Sin = multiplier * Math.sin(radians);
        newWave.Cos = multiplier * Math.cos(radians);
        newWave.Tan = multiplier * Math.tan(radians);
        newWave.Sinh = multiplier * Math.sinh(radians);
        newWave.Cosh = multiplier * Math.cosh(radians);
        newWave.Tanh = multiplier * Math.tanh(radians);

        return newWave;
    }
}
